### PR TITLE
ARC 1605 - Endpoint GitHubConfigurationGet removes the previous GH cloudconnection 

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- Added `gitHubAppId` to fetch the correct subscriptions in `github-configuration-get`
- Removed multiple DB calls.

**Why**
- The previous usage of `Subscription.getAllForHost` was pulling both cloud and server apps. This caused the cloud apps to be failed installations because they don't exist in the GHES. So, it was being removed from the DB as failed installations are all being removed. 

**How has this been tested?**  
- In local and ddev environment

### CODE REVIEW

File: src/routes/github/configuration/github-configuration-get.ts [hunk 1]
```diff
@@ -56,8 +56,11 @@ const installationConnectedStatus = async (
 	reqLog: Logger,
 	gitHubAppId?: number
 ): Promise<MergedInstallation[]> => {
-	const subscriptions = await Subscription.getAllForHost(jiraHost);
+	const subscriptions = await Subscription.getAllForHost(jiraHost, gitHubAppId);
 	const installationsWithSubscriptions = await getInstallations(subscriptions, reqLog, gitHubAppId);
+	await removeFailedConnectionsFromDb(reqLog, installationsWithSubscriptions, jiraHost);
+	reqLog.debug(`Removed failed installations`);
```
Comment (rachellerathbone) [inline src/routes/github/configuration/github-configuration-get.ts:62]: Should probably just call this logger as this is how it's typically approached in the code when passing req.log

Comment (rachellerathbone) [reply src/routes/github/configuration/github-configuration-get.ts:62]: Nit: don't need backticks here as we're not passing any values in.

File: src/routes/github/configuration/github-configuration-router.test.ts [hunk 2]
```diff
@@ -125,22 +125,6 @@ describe("Github Configuration", () => {
 
 			githubUserTokenNock(sub.gitHubInstallationId);
 
-			githubNock
```
Comment (rachellerathbone) [inline src/routes/github/configuration/github-configuration-router.test.ts:128]: Why have we simply removed these tests?

Comment (krazziekay) [reply src/routes/github/configuration/github-configuration-router.test.ts:128]: Reverted back and just removed `twice`. These calls are not made twice anymore, cause of the removal of the duplicated lines in `github-configuration-get`

File: src/models/subscription.ts [hunk 3]
```diff
@@ -41,9 +41,10 @@ export class Subscription extends Model {
 	repositoryStatus?: TaskStatus;
 	gitHubAppId?: number;
 
-	static async getAllForHost(jiraHost: string): Promise<Subscription[]> {
+	static async getAllForHost(jiraHost: string, gitHubAppId?: number): Promise<Subscription[]> {
```
Comment (gxueatlassian) [inline src/models/subscription.ts:44]: Hey @krazziekay , can you please change it to `gitHubAppId: number | undefined`?

Comment (krazziekay) [reply src/models/subscription.ts:44]: `getAllForHost` is already being used in a lot of places and if we define it like this, then the second parameter will be required. Thats why I made the second params as optional...

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**